File: Leetcode 3731 Find Missing Elements.py
# 2026/08/17 

class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        ans = []
        max_n = max(nums)
        min_n = min(nums)
        freq = [0] * (max_n - min_n + 1)
        if (max_n - min_n + 1) == len(nums):
            return []

        for i in nums:
            freq[i-min_n] = 1
        for index, j in enumerate(freq):
            if j == 0:
                ans.append(index+min_n)

        return ans

File: Leetcode 3731 Find Missing Elements.go
// 2026/08/17

func findMissingElements(nums []int) []int {
    var ans []int
    sliceMin := slices.Min(nums)
    sliceMax := slices.Max(nums) 
    seen := make([]int, (sliceMax-sliceMin)+1)
    if len(nums) == (sliceMax-sliceMin+1){
        return ans
    }

    for _, n := range nums {
        seen[n-sliceMin] = 1
    }
    for i:=0; i<len(seen);i++{
        if seen[i] == 0{
            ans = append(ans, i+sliceMin)
        }
    }
    return ans
}
